framework/auth/views.py

```diff
@@ -247,13 +392,15 @@ def send_confirm_email(user, email):
         merge_target = None
 
     campaign = campaigns.campaign_for_user(user)
-    # Choose the appropriate email template to use
+    # Choose the appropriate email template to use and add existing_user flag if a merge or adding an email.
     if merge_target:
         mail_template = mails.CONFIRM_MERGE
+        confirmation_url += '?existing_user=True'
```

Comment:
This doesn't really seem like something that should be put in a url. Could you not just check for this when you have to, including when you pass data to the template that uses it?